libtransmission/torrent-files.cc

```diff
@@ -307,18 +307,28 @@ void tr_torrent_files::remove(std::string_view parent_in, std::string_view tmpdi
 namespace
 {
 
+[[nodiscard]] bool isUnixReservedFile(std::string_view in) noexcept
```

Comment:
Won't this, `isUnixReservedChar` and their Windows counterparts appear unused?
Maybe you should wrap them in `#ifdef` accordingly?

Reply:
Well, my thoughts (added recently in a previous comment):
>I still decided to make distinct functions for Win32 and Unix reserved names, so that we can eventually use either ones depending on the filesystem in the future.

So I would prefer keeping them instead of removing them from the builds.

Reply:
I understand that, but I think compiler/clang-tidy will complain about it right now.

Reply:
How about this instead?

```suggestion
[[nodiscard, maybe_unused]] bool isUnixReservedFile(std::string_view in) noexcept
```

Reply:
Huh, I thought `[[maybe_unused]]` is for returned result, not for the function itself.
Also, `[[nodiscard]]` with `[[maybe_unused]]` looks pretty suspicious :)

Reply:
> `[[maybe_unused]]` looks pretty suspicious :)

I know 🤣

Reply:
I've verified the doc: it's valid for the function itself (https://en.cppreference.com/w/cpp/language/attributes/maybe_unused).

d30de1a93df7e188ae9c1384de3f5521ee3500d2